geograypher/utils/geometric.py

```diff
@@ -138,3 +139,116 @@ def projection_onto_plane(v1, e1, e2):
     out_of_plane_projection = orthogonal_projection(plane_normal, v1)
     in_plane_projection = v1 - out_of_plane_projection
     return in_plane_projection
+
+
+def clip_line_segments(
+    boundaries: typing.Tuple[pv.PolyData, pv.PolyData],
+    origins: np.ndarray,
+    directions: np.ndarray,
+    image_indices: typing.Union[typing.List[int], np.ndarray],
+    ray_limit: typing.Optional[float] = None,
+) -> typing.Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    """
+    Clips line segments between two boundary surfaces using ray tracing,
+    keeping only the segments that intersect both surfaces.
+
+    Args:
+        boundaries (Tuple[pv.PolyData, pv.PolyData]):
+            A tuple containing two PyVista PolyData surfaces representing the boundaries.
+        origins (np.ndarray):
+            Array of ray origin points (shape: [N, 3]).
+        directions (np.ndarray):
+            Array of ray direction vectors (shape: [N, 3]).
+        image_indices (List[int]):
+            List of indices associated with each ray, used for identification.
+            Tracks which image each ray corresponded to.
+        ray_limit (Optional[float], optional):
```

Comment:
nitpick: if there were a variable name the conveyed that rays were excluded for being too long rather than clipped that could be more clear, but I'm not sure how to convey that without being verbose.

Reply:
I can't think of one either, I tried to make it a little more clear in the docstring